src/filetree.rs

```diff
@@ -68,16 +68,25 @@ const DEFAULT_FILE_MODE: u32 = 0o700;
 use crate::sha512string::SHA512String;
 
 /// Metadata for a single file
-#[derive(Clone, Serialize, Deserialize, Debug, Hash, PartialEq)]
+#[derive(Clone, Serialize, Deserialize, Debug)]
 #[serde(rename_all = "kebab-case")]
 pub(crate) struct FileMetadata {
+    /// File source path
```

Comment:
Can you explain a bit more what this is, e.g. include an example and also a rationale for why we need this new field?